### PR TITLE
feat: Verifiy-only mode (pubkey-only)

### PR DESCRIPTION
**Pull request by maslennikov**
Implementation of verify-only mode, discussed in https://github.com/fastify/fastify-jwt/issues/273.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by maslennikov** [comment]
@mcollina good! i'll proceed with changes to the docs then

**Comment by maslennikov** [comment]
@mcollina code, types, tests, and docs are ready

**Comment by maslennikov** [comment]
Hello @mcollina! Are there any issues with this PR? Does it need some extra work to get a go?

**Review by mcollina — APPROVED** [review]
lgtm

File: README.md [hunk 1]
```diff
@@ -117,6 +117,17 @@ In this object `{ private, public }` the `public` key is a string or buffer cont
...
+#### Verify-only mode
+
+In cases when your incoming JWT tokens are issued by some trusted external
+service, and you need only to verify their signature without issuing, there is
+an option to configure `fastify-jwt` in *verify-only* mode by passing the
+`secret` object containing only a public key: `{ public }`.
+
+When only a public key provided, decode and verification functions will work as
+described below, but an exception will be thrown at an attempt to use any form
+of `sign` functionality.
```
Comment (Fdawgs) [inline README.md:129]:
```suggestion
In cases where your incoming JWT tokens are issued by a trusted external
service, and you need only to verify their signature without issuing, there is
an option to configure `fastify-jwt` in *verify-only* mode by passing the
`secret` object containing only a public key: `{ public }`.

When only a public key is provided, decode and verification functions will work as
described below, but an exception will be thrown at an attempt to use any form
of `sign` functionality.
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**